combat-trainer.lic

```diff
@@ -1280,6 +1282,22 @@ class TrainerProcess
     end
     DRRoom.dead_npcs.size > dead_count
   end
+  
+  def analyze(game_state)
+    return if game_state.npcs.empty?
+    case bput('analyze', 'roundtime', 'Analyze what', 'Your analysis reveals a massive opening',/You reveal an? .* [weakness|opening]/,/Your analysis reveals an? .* [weakness|opening]/, 'You fail to find any holes', 'There is nothing else', 'Face what', 'You must be closer', 'You must be standing', 'Strangely, you don\'t feel like fighting right now', 'flying too high for you to attack') #unless 
+    when 'Analyze what'
+      fput('face next')
+      analyze(game_state)
+    when 'You must be closer'
+      fput('engage')
+    when 'Your analysis reveals a massive opening'
+      return
+    when /You reveal an? .* [weakness|opening]/,/Your analysis reveals an? .* [weakness|opening]/, 'You fail to find any holes'
+      analyze(game_state)
```

Comment:
This recursive call means that you can potentially analyze many times in a row. Do you find that preferable to doing one analyze? Will this loop indefinitely if you are unable to analyze a mob (i.e. it's too hard)?